--- oniguruma-6.6.1/src/regcomp.c.nullreg~	2017-11-29 13:10:01.563465258 +0100
+++ oniguruma-6.6.1/src/regcomp.c	2017-11-29 13:12:14.314755488 +0100
@@ -6540,19 +6540,21 @@ onig_new(regex_t** reg, const UChar* pat
 	  OnigErrorInfo* einfo)
 {
   int r;
+  regex_t *regg;
 
-  *reg = (regex_t* )xmalloc(sizeof(regex_t));
-  if (IS_NULL(*reg)) return ONIGERR_MEMORY;
+  regg = (regex_t* )xmalloc(sizeof(regex_t));
+  if (IS_NULL(regg)) return ONIGERR_MEMORY;
 
-  r = onig_reg_init(*reg, option, ONIGENC_CASE_FOLD_DEFAULT, enc, syntax);
+  r = onig_reg_init(regg, option, ONIGENC_CASE_FOLD_DEFAULT, enc, syntax);
   if (r != 0) goto err;
 
-  r = onig_compile(*reg, pattern, pattern_end, einfo);
+  r = onig_compile(regg, pattern, pattern_end, einfo);
   if (r != 0) {
   err:
-    onig_free(*reg);
-    *reg = NULL;
+    onig_free(regg);
+    regg = NULL;
   }
+  *reg = regg;
   return r;
 }
 
